sources/osgShader/Compiler.js

```diff
@@ -748,13 +743,22 @@ define( [
 
                 shadowTexture = shadowTextures[ k ];
                 if ( shadowTexture ) {
-                    tex = this.getOrCreateSampler( 'sampler2D', shadowTexture.getName() );
-                    inputs.shadowTexture = tex;
+                    var shadowTexSamplerName = 'ShadowTexture' + k;
+
 
                     // per texture uniforms
                     var shadowTextureUniforms = this.getOrCreateTextureStateAttributeUniforms( shadowTexture, 'shadowTexture', k );
+
+                    // TODO: UGLY REMOVAL
+                    // create a specific texture getuniform for compiler
+                    shadowTextureUniforms[ 'shadowTexture' + shadowTexSamplerName ] = undefined;
```

Comment:
ShadowTexture is the first Texture inherited class with uniform attached.
So when we get those for the compiler, we get uniform we don't want (the int 
texture location for shader texture pointer to bind)

That Int uniform conflicts with the sample2D uniform in our compiler.

Either we make compiler handle that case (not easy) or we make sure we don't do those conflicts (meaning a special getOrCreateUniform, with a parameter like "getShaderTextureLocation" flag at false when called from compiler, true when called from other code)